Snowplow AWS pipeline support (#655)
objectiv/objectiv-analytics: backend/objectiv_backend/snowplow/docs/gcp.md

# Configuring the Objectiv Collector to work with GCP
The Objectiv Collector can be configured to work with Snowplow on GCP. The Snowplow GCP setup uses GCP PubSub topics 
(message queue) to connect various stages in the pipeline. The setup works as follows:
- Events arrive at the Objectiv Collector, and are validated.
- Good events are published on the `raw` topic on PubSub (which is read by the Enrich process)
- Bad events (invalid) are published on the `bad` topic on PubSub.

Before starting up the Objectiv Collector, some preparation is required. The following sections assume there's already
a running and functional Snowplow pipeline setup on GCP. If you don't, Snowplow has a 
[quick start installation guide](https://docs.snowplowanalytics.com/docs/open-source-quick-start/quick-start-installation-guide-on-gcp/)
to get you up and running.

> #### Preparation

Author: yes, simply removed, as it is kind of redundant

Before starting up the Objectiv Collector, some preparation is required. The following sections assume there's already
a running and functional Snowplow pipeline setup on GCP.

#### Preparation
To be able to push events into the message queue the following needs to be set up:
- A GCP service account (`json`) with permission to write to PubSub
- The `id` of the GCP project that holds the PubSub topic
- The `id` of the PubSub raw topic (message queue)
- The `id` of the PubSub bad topic (message queue)

These will be used to configure the collector in the next step.

#### Starting the collector
The configuration for the collector is controlled through environment variables. They allow you to configure which outputs
will be used. Settings specific to the PubSub sin are:

- `SP_GCP_PROJECT` This is the `id` of the project on GCP where the PubSub topic is located
- `SP_GCP_PUBSUB_TOPIC_RAW` This is the `id` of the PubSub topic to publish events to
- `SP_GCP_PUBSUB_TOPIC_BAD` This is the `id` of the PubSub topic to publish bad/invalid events to
- `GOOGLE_APPLICATION_CREDENTIALS` This is the path to a `json` containing a service account on GCP that allows publishing
 to the PubSub topic.

> Additionally, the collector checks `SP_GCP_PROJECT`. If it is set, the snowplow sink is automatically enabled.

Reviewer: Can be simpler
```suggestion
If these are set, the snowplow sink is automatically enabled.
```

#### Using docker-compose
To run this setup in docker, make sure that the aforementioned environment variables are properly set and available in the
container. Also take care that the path to the credentials is actually available in the container.

When using `docker-compose`, the following yaml snippet would do the trick:
```yaml
  objectiv_collector:
    container_name: objectiv_collector
    depends_on:

>       - objectiv_postgres

Reviewer: nein nein nein?
